sql-plugin/src/main/scala/com/nvidia/spark/rapids/SortUtils.scala

```diff
@@ -226,15 +237,16 @@ class GpuSorter(
           batches.foreach { cb =>
             tabs += GpuColumnVector.from(cb)
           }
-          // In the current version of cudf merge does not work for structs or lists (nested types)
+          // In the current version of cudf merge does not work for lists
           // This should be fixed by https://github.com/rapidsai/cudf/issues/8050
-          val hasNested = {
+          val hasUnsupportedNested = {
             val tab = tabs.head
             (0 until tab.getNumberOfColumns).exists { i =>
-              tab.getColumn(i).getType.isNestedType
+              // Map is list of struct in cudf, so checking list type covers the map type.
+              isOrHasListType(tab.getColumn(i))
```

Comment:
Walking the cudf type is way more expensive, as we have to traverse the JNI and track closing native resources.  Instead we can leverage the Spark type and `TrampolineUtil.dataTypeExistsRecursively`, e.g.:
```scala
          val hasUnsupportedNested = (0 until batches.head.numCols()).exists { i =>
            TrampolineUtil.dataTypeExistsRecursively(batches.head.column(i).dataType(),
              t => t.isInstanceOf[ArrayType] || t.isInstanceOf[MapType])
          }
```

Reply:
Updated